torch/_inductor/codegen/cpp.py

```diff
@@ -1209,8 +1210,38 @@ def scale_index_with_offset(
         new_index = sympy_subs(index, replacement)
         return new_index
 
-    @staticmethod
-    def indirect_indexing(index_var, size, check=True):
+    def index_to_str(self, index: sympy.Expr) -> str:
+        """
+        Convert an index expr to a string that can be used in triton code.
+        e.g. a sympy expression "s2" may actually appear as "ks1" in the cpp kernel.
+        """
+        return cexpr(self.rename_indexing(index))
+
+    def indirect_indexing(self, index_var, size, check=True):
+        generate_assert = (
+            check or config.debug_index_asserts
+        ) and config.cpp.assert_indirect_indexing
```

Comment:
As discussed before, let's just have one way to deactivate the asserts in both CPU and CUDA.

Then, you can lift all this code to `common` rather than duplicating it, and just have an op in each of the codegen's that tells you whether to use `TORCH_CHECK` or `tl.device_assert`.